contracts/contracts/validator-registry/rewards/RewardManager.sol

```diff
@@ -0,0 +1,202 @@
+        require(newClaimAddress != address(0) && newClaimAddress != msg.sender, InvalidAddress());
+        overriddenClaimAddresses[msg.sender] = newClaimAddress;
+        emit OverrideClaimAddressSet(msg.sender, newClaimAddress);
+    }
+
+    /// @dev Removes the override claim address for a reward recipient.
+    function removeOverriddenClaimAddress() external whenNotPaused {
+        overriddenClaimAddresses[msg.sender] = address(0);
+        emit OverrideClaimAddressRemoved(msg.sender);
+    }
+
+    /// @dev Allows a reward recipient to claim their rewards.
+    function claimRewards() external whenNotPaused {
+        uint256 amount = unclaimedRewards[msg.sender];
+        require(amount > 0, NoRewardsToClaim());
+        unclaimedRewards[msg.sender] = 0;
+        (bool success, ) = payable(msg.sender).call{value: amount}("");
+        if (!success) {
+            revert RewardsClaimFailed();
+        }
+        emit RewardsClaimed(msg.sender, amount);
+    }
+
+    /// @dev Allows the owner to claim orphaned rewards to appropriate addresses.
+    function claimOrphanedRewards(bytes[] calldata pubkeys, address toPay) external onlyOwner {
+        uint256 totalAmount = 0;
+        uint256 len = pubkeys.length;
+        for (uint256 i = 0; i < len; ++i) {
+            bytes calldata pubkey = pubkeys[i];
+            uint256 amount = orphanedRewards[pubkey];
+            require(amount > 0, NoOrphanedRewards());
+            orphanedRewards[pubkey] = 0;
+            totalAmount += amount;
+        }
+        (bool success, ) = payable(toPay).call{value: totalAmount}("");
+        if (!success) {
+            revert OrphanedRewardsClaimFailed();
+        }
+    }
+
+    /// @dev Allows the owner to set the vanilla registry address.
+    function setVanillaRegistry(address vanillaRegistry) external onlyOwner {
+        _setVanillaRegistry(vanillaRegistry);
+    }
+
+    /// @dev Allows the owner to set the mev commit avs address.
+    function setMevCommitAVS(address mevCommitAVS) external onlyOwner {
+        _setMevCommitAVS(mevCommitAVS);
+    }
+
+    /// @dev Allows the owner to set the mev commit middleware address.
+    function setMevCommitMiddleware(address mevCommitMiddleware) external onlyOwner {
+        _setMevCommitMiddleware(mevCommitMiddleware);
+    }
+
+    /// @dev Allows the owner to set the auto claim gas limit.
+    function setAutoClaimGasLimit(uint256 autoClaimGasLimit) external onlyOwner {
+        _setAutoClaimGasLimit(autoClaimGasLimit);
+    }
+
+    // solhint-disable-next-line no-empty-blocks
+    function _authorizeUpgrade(address) internal override onlyOwner {}
+
+    function _setVanillaRegistry(address vanillaRegistry) internal {
+        require(vanillaRegistry != address(0), InvalidAddress());
+        _vanillaRegistry = VanillaRegistryStorage(vanillaRegistry);
+        emit VanillaRegistrySet(address(vanillaRegistry), vanillaRegistry);
```

Comment:
It is not working as intended from the event signature (`event VanillaRegistrySet(address indexed oldVanillaRegistry, address indexed newVanillaRegistry);`). We need to store old address in some temp variable in order to emit it or delete it from event signature.